lib/src/signed_video_openssl.c

```diff
@@ -575,19 +644,66 @@ openssl_set_hash_algo(void *handle, const char *name_or_oid)
 {
   openssl_crypto_t *self = (openssl_crypto_t *)handle;
   if (!self) return SVI_INVALID_PARAMETER;
+  // NULL pointer as input means default setting.
   if (!name_or_oid) {
     name_or_oid = DEFAULT_HASH_ALGO;
   }
 
-  ASN1_OBJECT *hash_algo_oid = OBJ_txt2obj(name_or_oid, 0 /* Accept both name and OID */);
-  const EVP_MD *hash_algo_type = EVP_get_digestbyobj(hash_algo_oid);
-  if (!hash_algo_type) {
-    DEBUG_LOG("Could not identify hashing algorithm: %s", name_or_oid);
-    return SVI_INVALID_PARAMETER;
+  svi_rc status = SVI_UNKNOWN;
+  SVI_TRY()
+    ASN1_OBJECT *hash_algo_obj = OBJ_txt2obj(name_or_oid, 0 /* Accept both name and OID */);
+    SVI_THROW_IF_WITH_MSG(!hash_algo_obj, SVI_INVALID_PARAMETER,
+        "Could not identify hashing algorithm: %s", name_or_oid);
+    SVI_THROW(obj_to_oid_and_type(&self->hash_algo, hash_algo_obj));
+    DEBUG_LOG("Setting hash algo %s that has ASN.1/DER coded OID length %zu", name_or_oid,
+        self->hash_algo.encoded_oid_size);
+  SVI_CATCH()
+  SVI_DONE(status)
+
+  return status;
+}
+
+svi_rc
+openssl_set_hash_algo_by_encoded_oid(void *handle,
+    const unsigned char *encoded_oid,
+    size_t encoded_oid_size)
+{
+  openssl_crypto_t *self = (openssl_crypto_t *)handle;
+  if (!self || !encoded_oid || !encoded_oid_size) return SVI_INVALID_PARAMETER;
```

Comment:
nit: I have seen that size is checked differently in other parts of the code. "!encoded_oid_size" works fine but check this out so it's consistent in the code.